### ISSUE TITLE
Subscribe: Red asterisk * is not sufficient to indicate the field is required field

### ISSUE THREAD

**Issue by XiuJuanXia** [mmiron-HC/WST-RSAMS#104]
URL: https://hcdev.openplus.ca/en/form/subscribe
User Agent: Chrome
Language: Both

Issue Description: When users leave three fields that have red asterisk * empty and activate the “Submit” button, error message appears and users cannot submit the form. Word “required” should be part of the label so that all users will be aware that these fields are mandatory fields before they try to submit the form without entering data or making selections for these fields. Please refer to the WET form validation: https://wet-boew.github.io/wet-boew/demos/formvalid/formvalid-en.html
![image](https://user-images.githubusercontent.com/88776404/136954462-db20881f-e338-40d1-8f6e-ff129f0315dc.png)

Steps to reproduce:

1. Navigate through the forms and leave the forms empty
2. Activate the "Submit" button
3. Error message appears.

Expected results: Word "required" should be part of the label so that all users will be aware that these fields are mandatory fields before they try to submit the form without entering data or making selections for these fields.

**Comment by echouin** [comment]
@XiuJuanXia this is the same as Issue #16: This is a WxT / Drupal core issue. There is an issue for it in Drupal. This is tagged as moderate, so I ﻿assume not a deal breaker

**Comment by XiuJuanXia** [comment]
@echouin tagged as moderate.

**Comment by echouin** [comment]
From OP: This is done and will be available on the site in the next pull
﻿

**Comment by XiuJuanXia** [comment]
@echouin The word "required" is included as part of the label for the required fields. However the screen reader users will hear the word "required" twice because there is a hidden "required". Here is the source code: `<span class="fieldset-legend js-form-required form-required">Select your preferred language of correspondence <span class="required">(required)</span><span aria-hidden="true" class="hidden"> (required)</span></span>`. Please remove code `<span aria-hidden="true" class="hidden"> (required)</span>` for all the required fields in English and French . The visual text is sufficient for all users.

**Comment by echouin** [comment]
@XiuJuanXia decision required. From OP:
<span aria-hidden="true" class="hidden"> (required)</span> is default Drupal code and we don't want to mess with it.
﻿
﻿Recommendations:
﻿
﻿1. Live with it. It's a moderate issue.
﻿2. We could remove (required) from the label, and have HTML at the top of the page that says "fields marked with an asterisk (*) are required.

Eric comment: I think Option 2 would work.
﻿
**Please let me know how you would like to proceed?**

**Comment by XiuJuanXia** [comment]
> @XiuJuanXia decision required. From OP: (required) is default Drupal code and we don't want to mess with it. ﻿ ﻿Recommendations: ﻿ ﻿1. Live with it. It's a moderate issue. ﻿2. We could remove (required) from the label, and have HTML at the top of the page that says "fields marked with an asterisk (*) are required.
> 
> Eric comment: I think Option 2 would work. ﻿ **Please let me know how you would like to proceed?**

Yes. Option 2 would work. - I will have to take back my comments and need to discuss with the team to see if the option 2 is sufficient.

**Comment by sarasuri** [comment]
@echouin 
Required form fields are identified with red asterisks. However, the asterisks are added to field’s label using CSS styles and when the page is viewed disabling the CSS styles, the asterisks are not available. This is a failure under WCAG SC 1.3.1 Info and Relationships   (https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/F87).

As per comment above, if the (required) is default Drupal code to identify required fields we noticed the code is not consistently applied for all required fields in the system. `<span` aria-hidden="true" class="hidden"> (required)</span>`
Further, on the French page, text “required” is in English. For example, on the “Subscribe” page, the first three fields are required but the first one doesn’t have that default code.

We would like to go with **option 1** as we think the option 2 wouldn’t work for the following reasons. 

1. Currently, the (required) default Drupal code is marked up in a way it is hidden from visual display as well as from assistive technologies such as screen reader. 
2. Since the asterisk is not available to screen readers, when the meaning of the asterisk defined at the start of the form "fields marked with an asterisk (*) are required”, it may confuse screen reader users unless it is hidden from screen readers.
3. For the second and third required fields, with option 2 the required status will not be available to screen readers as the visually required status is specified in the title (legend) of the group using only the asterisk.

![image](https://user-images.githubusercontent.com/12256067/137519650-5dabe508-2409-4d83-8ca1-c0b63a4f7d70.png)


![image](https://user-images.githubusercontent.com/12256067/137519697-02ef7b3d-bce2-45fd-b363-aec6af4c219f.png)

**Comment by echouin** [comment]
@sarasuri @mmiron-HC from OP: Done.

**Comment by hydegeoff** [comment]
The issue is fixed. Issue closed.
